### PR TITLE
[Rearch] Migrate regression task #1

### PR DESCRIPTION
**Pull request by letitz**
Move a couple things out of `utask_main` into `utask_preprocess` and `utask_postprocess`, add a bunch of tests.

@jonathanmetzman PTAL. I would appreciate some guidance on the TODOs in unit tests, specifically around how to set up the environment so that we do not need to mock out so much code.

### CODE REVIEW

File: src/clusterfuzz/_internal/bot/tasks/utasks/regression_task.py [hunk 1]
```diff
@@ -206,50 +208,30 @@ def validate_regression_range(testcase, testcase_file_path, job_type,
...
 
 
-def find_regression_range(testcase_id, job_type):
+def find_regression_range(
+    testcase_id: str,
+    testcase: uworker_io.UworkerEntityWrapper,
+    job_type: str,
+    setup_input: uworker_io.SetupInput,
+    task_input: uworker_io.RegressionTaskInput,
+) -> Optional[uworker_io.UworkerOutput]:
```
Comment (jonathanmetzman) [inline src/clusterfuzz/_internal/bot/tasks/utasks/regression_task.py:217]: I would probably just pass uworker_input in, instead of these.

**Review by jonathanmetzman — APPROVED** [review]
LGTM

**Review by jonathanmetzman — COMMENTED** [review]
Err...mostly LGTM

**Comment by jonathanmetzman** [comment]
Conflict. Please fix.

**Comment by letitz** [comment]
Rebased.

File: src/clusterfuzz/_internal/bot/tasks/utasks/regression_task.py [hunk 2]
```diff
@@ -16,6 +16,8 @@
 
 import random
 import time
+from typing import Dict
+from typing import Optional
```
[the remark this answers is not in the document]
Comment (jonathanmetzman) [reply src/clusterfuzz/_internal/bot/tasks/utasks/regression_task.py:20]: This hasn't been fixed yet.

**Comment by jonathanmetzman** [comment]
Can you rebase again?
Merging this is going to be my top priority tomorrow.

File: src/clusterfuzz/_internal/bot/tasks/utasks/regression_task.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (letitz) [reply src/clusterfuzz/_internal/bot/tasks/utasks/regression_task.py:20]: Actually, `butler.py format` undoes my changes when I write `from typing import Dict, Optional` and reverts to this version.

**Comment by letitz** [comment]
Rebased!

**Comment by jonathanmetzman** [comment]
Will merge this tomorrow AM, had to fix some issues with postprocess today.
For the record, postprocess is now happening on another machine for analyze_task.

**Pull request merged**